scripts/bumper/main.go

```diff
@@ -375,10 +382,43 @@ func detectNewCommits(ctx context.Context, logger *logrus.Entry, stagingDir, cen
 			}
 		}
 	}
-	sort.Slice(commits, func(i, j int) bool {
-		return commits[i].Date.Before(commits[j].Date)
-	})
-	return commits, nil
+	// we would like to intertwine the commits from each upstream repository by date, while
+	// keeping the order of commits from any one repository in the order they were committed in
+	var orderedCommits []commit
+	indices := map[string]int{}
+	for repo := range commits {
+		indices[repo] = 0
+	}
+	for {
+		// find which repo's commit stack we should pop off to get the next earliest commit
+		nextTime := time.Now()
+		var nextRepo string
+		for repo, index := range indices {
+			if commits[repo][index].Date.Before(nextTime) {
+				nextTime = commits[repo][index].Date
+				nextRepo = repo
+			}
+		}
+
+		// pop the commit, add it to our list and do housekeeping for our index records
+		orderedCommits = append(orderedCommits, commits[nextRepo][indices[nextRepo]])
+		if indices[nextRepo] == len(commits[nextRepo])-1 {
+			delete(indices, nextRepo)
+		} else {
+			indices[nextRepo] += 1
+		}
+
+		if len(indices) == 0 {
+			break
+		}
+	}
+
+	// our ordered list is descending, but we need to cherry-pick from the oldest first
+	var reversedCommits []commit
+	for i := range orderedCommits {
+		reversedCommits = append(reversedCommits, orderedCommits[len(orderedCommits)-i-1])
+	}
```

Comment:
No easier way to reverse a list (array)?

Reply:
What is difficult about this, or, how would you want to improve it?

Reply:
```
slices.Reverse(orderedCommits)
```
Might need to use `golang.oef/x/exp/slices` in 1.20; `slices` is part of the standard library in 1.21

Reply:
Or even do it in place rather than doing an `append`. Unless you need to keep `orderedCommits` around unchanged?

Reply:
Eh, it's a short-lived CLI tool. The extra memory doesn't really matter. Happy to use the slices API when it lands in the stdlib.